public/app/features/variables/inspect/VariablesUnknownButton.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { FC, MouseEvent, useCallback, useMemo, useState } from 'react';
+import { Provider, useSelector } from 'react-redux';
+import { IconButton, Modal } from '@grafana/ui';
+
+import { StoreState } from '../../../types';
+import { getVariables } from '../state/selectors';
+import { createUsagesNetwork, transformUsagesToNetwork } from './utils';
+import { NetWorkGraph } from './NetworkGraph';
+import { store } from '../../../store/store';
+import { VariableIdentifier } from '../state/types';
+
+interface OwnProps {
+  identifier: VariableIdentifier;
+}
+
+interface ConnectedProps {}
+
+interface DispatchProps {}
+
+type Props = OwnProps & ConnectedProps & DispatchProps;
+
+export const UnProvidedVariablesUnknownGraphButton: FC<Props> = ({ identifier }) => {
+  const [showModal, setShowModal] = useState(false);
+  const variables = useSelector((state: StoreState) => getVariables(state));
+  const dashboard = useSelector((state: StoreState) => state.dashboard.getModel());
+  const { unknown } = useMemo(() => createUsagesNetwork(variables, dashboard), [variables, dashboard]);
+  const network = useMemo(() => transformUsagesToNetwork(unknown).find(n => n.variable.id === identifier.id), [
+    identifier,
+    unknown,
+  ]);
+  const unknownExist = useMemo(() => Object.keys(unknown).length > 0, [unknown]);
+  const onClick = useCallback(
+    (event: MouseEvent) => {
+      event.preventDefault();
+      setShowModal(true);
+    },
+    [setShowModal]
+  );
+  const onClose = useCallback(() => setShowModal(false), [setShowModal]);
+
+  if (!unknownExist || !network) {
+    return null;
+  }
+
+  const nodes = network.nodes.map(n => {
```

Comment:
nit: It feels like there are some repetition between the different `Variables*Button` components. But I'm not sure it is worth trying to extract it to a shared component. That might just be adding complexity. WDYT?

Reply:
Yeah, I thought about this too. I might have a second look